src/hb-aat-layout-feat-table.hh

```diff
@@ -68,18 +93,59 @@ struct FeatureName
 				 * as the default. */
   };
 
+  inline unsigned int get_settings (const feat                      *feat,
+				    hb_aat_layout_feature_setting_t *default_setting,
+				    unsigned int                     start_offset,
+				    unsigned int                    *count,
+				    hb_aat_layout_feature_setting_t *settings) const
+  {
+    const UnsizedArrayOf<SettingName>& settings_table = feat+settingTableZ;
+    unsigned int settings_count = nSettings;
+    if (count && *count)
+    {
+      unsigned int len = MIN (settings_count - start_offset, *count);
+      for (unsigned int i = 0; i < len; i++)
+        settings[i] = settings_table[start_offset + i].get_setting ();
+      *count = len;
+    }
+    if (default_setting)
+    {
+      unsigned int index = (featureFlags & NotDefault) ? featureFlags & IndexMask : 0;
+      *default_setting = ((featureFlags & Exclusive) && index < settings_count)
+			 ? settings_table[index].get_setting ()
+			 : HB_AAT_LAYOUT_SELECTOR_INVALID;
+    }
+    return settings_count;
+  }
+
+  inline hb_aat_layout_feature_type_t get_feature_type () const
+  { return (hb_aat_layout_feature_type_t) (unsigned int) feature; }
+
+  inline hb_ot_name_id_t get_feature_name_id () const { return nameIndex; }
+
+  inline hb_ot_name_id_t get_feature_setting_name_id (const feat                      *feat,
```

Comment:
These functions taking feat table look weird.  I think things can be done differently, but I need to pull down the code and take a deeper look than just reading the patch.